### PR TITLE
feat: adds error based on status code for http instrumentation.

### PR DESCRIPTION
**Pull request by jcchavezs**
This PR makes it possible to to add errors based on status code for http instrumentations.

Ping @adriancole @mkante @DanielMSchmidt

### CODE REVIEW

File: packages/zipkin/src/instrumentation/httpServer.js [hunk 1]
```diff
@@ -85,6 +85,8 @@ class HttpServerInstrumentation {
     this.tracer.recordBinary('http.status_code', statusCode.toString());
     if (error) {
       this.tracer.recordBinary('error', error.toString());
+    } else if (statusCode < 200 || statusCode > 399) {
```
Comment (jcchavezs) [inline packages/zipkin/src/instrumentation/httpServer.js]: This is the piece of code doing the trick for http servers.

File: packages/zipkin-instrumentation-restify/src/restifyMiddleware.js [hunk 2]
```diff
@@ -36,12 +34,7 @@ module.exports = function restifyMiddleware({tracer, serviceName, port = 0}) {
         res.removeListener('finish', onCloseOrFinish);
 
         tracer.scoped(() => {
-          const error =
-            res.statusCode !== 0 && (res.statusCode < LOWER_LIMIT || res.statusCode > UPPER_LIMIT)
-            ? res.statusCode
-            : null;
-
-          instrumentation.recordResponse(id, res.statusCode, error);
+          instrumentation.recordResponse(id, res.statusCode);
```
Comment (jcchavezs) [inline packages/zipkin-instrumentation-restify/src/restifyMiddleware.js]: `restify` used to add the error code based on status code but now we moved the logic to the http server instrumentation.

File: packages/zipkin-instrumentation-express/src/expressMiddleware.js [hunk 3]
```diff
@@ -32,9 +32,7 @@ module.exports = function expressMiddleware({tracer, serviceName, port = 0}) {
 
       res.on('finish', () => {
         tracer.scoped(() => {
-          const error = res.statusCode >= 400 ? res.statusCode : null;
-
-          instrumentation.recordResponse(id, res.statusCode, error);
+          instrumentation.recordResponse(id, res.statusCode);
```
Comment (jcchavezs) [inline packages/zipkin-instrumentation-express/src/expressMiddleware.js]: Same https://github.com/openzipkin/zipkin-js/pull/261/files#r215305990

**Comment by adriancole** [comment]
thanks for fixing broken windows!

File: packages/zipkin/src/instrumentation/httpServer.js (hunk 1 above)
Comment (mkante) [reply packages/zipkin/src/instrumentation/httpServer.js]: Looks good to me

**Pull request merged**